Cura.proto

```diff
@@ -61,6 +61,23 @@ message Polygon {
     float line_width = 3; // The width of the line being laid down
 }
 
+message LayerOptimized {
+    int32 id = 1;
+    float height = 2; // Z position
+    float thickness = 3; // height of a single layer
+
+    repeated PathSegment path_segment = 4; // layer data
```

Comment:
That seems like a good idea!
Sending the extruder along to the frontend had still to be done anyway.
Visualizing the gcode differently depending on the extruder would be a very nice feature.